src/zepben/evolve/streaming/grpc/connect.py

```diff
@@ -29,40 +29,41 @@ def _secure_grpc_channel_builder(host: str = "localhost", rpc_port: int = 50051)
         .make_secure()
 
 
-def _grpc_channel_builder_from_password(client_id: str, username: str, password: str, host: str, rpc_port: int,
-                                        conf_address: Optional[str]) -> GrpcChannelBuilder:
-    token_fetcher: ZepbenTokenFetcher
-    if conf_address:
-        token_fetcher = create_token_fetcher(conf_address=conf_address)
-    else:
-        # noinspection PyArgumentList
-        token_fetcher = ZepbenTokenFetcher(
-            audience="https://evolve-ewb/",
-            issuer_domain="zepben.au.auth0.com",
-            auth_method=AuthMethod.AUTH0,
-            verify_certificate=True
-        )
+def _grpc_channel_builder_from_password(client_id: str, username: str, password: str, host: str, rpc_port: int, token_fetcher: ZepbenTokenFetcher
+                                        ) -> GrpcChannelBuilder:
     token_fetcher.token_request_data.update({
         'client_id': client_id,
         'username': username,
         'password': password,
         'grant_type': 'password',
         'scope': 'offline_access'
     })
-
     return GrpcChannelBuilder() \
         .socket_address(host, rpc_port) \
         .make_secure() \
         .token_fetcher(token_fetcher)
 
 
-def connect_secure(host: str = "localhost", rpc_port: int = 50051) -> grpc.aio.Channel:
+def connect_tls(host: str = "localhost", rpc_port: int = 50051) -> grpc.aio.Channel:
     return _secure_grpc_channel_builder(host, rpc_port).build()
 
 
 def connect_with_password(client_id: str, username: str, password: str, host: str = "localhost", rpc_port: int = 50051,
-                          conf_address: Optional[str] = None) -> grpc.aio.Channel:
-    return _grpc_channel_builder_from_password(client_id, username, password, host, rpc_port, conf_address).build()
+                          conf_address: Optional[str] = None, **kwargs) -> grpc.aio.Channel:
+    token_fetcher: ZepbenTokenFetcher
+    if conf_address:
+        token_fetcher = create_token_fetcher(conf_address=conf_address)
+    elif {"audience", "issuer_domain", "auth_method"} <= kwargs.keys():
```

Comment:
I learned this trick just now. It's a subset check

Reply:
that is cool